lib/postinst-migrations.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/sh
+#
+# Migrations from previous ubuntu-pro-client/ubuntu-advantage-tools versions
+#
+# These exist in this script separate from the *.postinst scripts because they
+# may need to be called from either ubuntu-pro-client.postinst or from
+# ubuntu-advantage-tools.postinst, depending on the situation.
+#
+# Because ubuntu-advantage-tools Depends on ubuntu-pro-client,
+# ubuntu-pro-client.postinst will always be executed before
+# ubuntu-advantage-tools.postinst.
+#
+# If the system already has ubuntu-pro-client and is upgrading to a new version
+# of ubuntu-pro-client, that means all migrations present in
+# ubuntu-advantage-tools.postinst must have already run at the time the system
+# upgraded to the renamed ubuntu-pro-client. That means the migrations in this
+# file can and should execute as part of ubuntu-pro-client.postinst.
+#
+# If upgrading from before the rename to the current version (from before
+# version 31), then not necessarily all migrations inside of
+# ubuntu-advantage-tools.postinst will have already run. Because the migrations
+# present in this file may depend on those previous migrations, then this file
+# must execute at the end of ubuntu-advantage-tools.postinst.
+#
+# In practice, this file is executed conditionally in either
+# ubuntu-pro-client.postinst or ubuntu-advantage-tools.postinst using version
+# checks. If we're upgrading from a version earlier than 31, then this executes
+# in ubuntu-advantage-tools.postinst. If we're upgrading from version 31 or
+# later, then this executes in ubuntu-pro-client.postinst.
```

Comment:
one post inst is comparing with `lt "31~"`. The other, with `gt "30"`. Is this accurate? If so, why are 30.x versions special (this should either be added to this docs or have the version constraints fixed).

Reply:
I just wrote them at different times. This works as is because there is no `30.x` version, but you are right that it would be more correct to have `gt "31~"` - I'll  change it